File: food-delivery-dotnet/restaurant-service/Models/MenuItem.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace RestaurantService.Models
{
    public class MenuItem
    {
        public int Id { get; set; }
        
        [Required]
        [StringLength(100)]
        public string Name { get; set; } = string.Empty;
        
        [StringLength(500)]
        public string Description { get; set; } = string.Empty;
        
        [Column(TypeName = "decimal(10,2)")]
        public decimal Price { get; set; }
        
        [StringLength(50)]
        public string Category { get; set; } = string.Empty;
        
        public bool IsAvailable { get; set; } = true;
        
        public int RestaurantId { get; set; }
        
        public virtual Restaurant Restaurant { get; set; } = null!;
        
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    }
}

File: food-delivery-ruby/restaurant-service/models/menu_item.rb
require 'activerecord'

class MenuItem < ActiveRecord::Base
  belongs_to :restaurant
  validates :name, presence: true
  validates :price, presence: true
  validates :restaurant_id, presence: true

  def as_json(options = {})
    {
      id: id,
      restaurant_id: restaurant_id,
      name: name,
      description: description,
      price: price,
      category: category,
      available: available,
      created_at: created_at,
      updated_at: updated_at
    }
  end
end